ecs-cli/modules/clients/aws/iam/client.go

```diff
@@ -81,12 +81,15 @@ func (c *iamClient) CreatePolicy(input iam.CreatePolicyInput) (*iam.CreatePolicy
 }
 
 // CreateOrFindRole returns a new role ARN or an empty string if role already exists
-func (c *iamClient) CreateOrFindRole(roleName, roleDescription, assumeRolePolicyDoc string) (string, error) {
+func (c *iamClient) CreateOrFindRole(roleName, roleDescription, assumeRolePolicyDoc string, tags []*iam.Tag) (string, error) {
 	createRoleRequest := iam.CreateRoleInput{
 		AssumeRolePolicyDocument: aws.String(assumeRolePolicyDoc),
 		Description:              aws.String(roleDescription),
 		RoleName:                 aws.String(roleName),
 	}
+	if len(tags) > 0 {
+		createRoleRequest.Tags = tags
+	}
 	roleResult, err := c.CreateRole(createRoleRequest)
 	// if err is b/c role already exists, OK to continue
 	if err != nil && !utils.EntityAlreadyExists(err) {
```

Comment:
What happens if we try to apply tags to an existing role? Does it return a distinct error or succeed? If the former, wondering if we should be checking for a specific error here & allowing it to succeed (like with `EntityAlreadyExists`) or let it fail.

Reply:
Adding here what we discussed: trying to re-create an existing role but with tags will still fail with an `EntityAlreadyExists` error.

Reply:
I'm happy to go along with what @allisaurus agrees with but curious why this is the correct behavior?

Reply:
As per offline discussion, I understand now that this is strictly on creation of a new role, so this seems fine.